Rubberduck.Parsing/VBA/ParseCoordinator.cs

```diff
@@ -116,19 +202,17 @@ private void Cancel(bool createNewTokenSource = true)
         }
 
         /// <summary>
-        /// For the use of tests only
+        /// For the use of tests & reflection API only
         /// </summary>
-        /// 
         public void Parse(CancellationTokenSource token)
         {
             SetSavedCancellationTokenSource(token);
             ParseInternal(token.Token);
         }
```

Comment:
This is broken now. Do we really need to pass the token for the API? Otherwise, this method should only exist on the `TestParseCoordinator`. For this to work, the test version must initiate the parse without canceling inside the lock.

Reply:
Moreover, should the API really use the version without logging?

Reply:
I was surprised to see API using it when I checked the usages as I meant to move that to the test coordinator especially as it contradicted the original comment expressing the intended usage of that method. In API's case, we support both `BeginParse` and `Parse` so that API can run synchronously if they choose to. 

The API also can receive events from parser state, so it should be possible for the API to request a cancellation. 

That said, you're right - API really should be logged anyway. That bit needs to be refactored a bit more.